bbhash_iter_test.go

```diff
@@ -13,138 +13,70 @@ import (
-	bufSz := 128
-
-	r := strings.NewReader(input)
-	chunks := bbhash.ReadChunks(r, bufSz)
-	gotKeyHashes := bbhash.Keys(hashFunc, chunks)
-
-	wantChunks := slices.Collect(slices.Chunk([]byte(input), bufSz))
-	wantKeyHashes := []uint64{}
-
-	// Last hash element might containt zeroes, meaning we have to append some zeroes  to the original input
-	// so that both will produce the same hash
-	if len(wantChunks[len(wantChunks)-1]) != bufSz {
-		for i := len(wantChunks[len(wantChunks)-1]); i < bufSz; i++ {
-			wantChunks[len(wantChunks)-1] = append(wantChunks[len(wantChunks)-1][:], 0)
-		}
-	}
-
-	// Hash the keys using fast.Hash64() using 1 as the seed, which was chosen arbitrarily, to compare to the keys from Keys()
-	for i := range wantChunks {
-		wordHash := fast.Hash64(1, []byte(wantChunks[i]))
-		wantKeyHashes = append(wantKeyHashes, wordHash)
-	}
-
-	for i := range wantKeyHashes {
-		if diff := cmp.Diff(gotKeyHashes[i], wantKeyHashes[i]); diff != "" {
-			t.Errorf("Keys() (-got +want) \n%s", diff)
-		}
-
-	}
+var fastHashFunc = func(buf []byte) uint64 {
+	return fast.Hash64(uint64(123), buf)
 }
 
-func TestHashKeysFromChunksLongerWordsSHA256(t *testing.T) {
-	// String taken from https://www.lipsum.com/
-	input := "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum."
-	hashFunc := 1 // 0 = Fasthash, 1 = sha256
-	bufSz := 128
-
-	r := strings.NewReader(input)
-	chunks := bbhash.ReadChunks(r, bufSz)
-	gotKeyHashes := bbhash.Keys(hashFunc, chunks)
-
-	wantChunks := slices.Collect(slices.Chunk([]byte(input), bufSz))
-	wantKeyHashes := []uint64{}
-
-	// Last hash element might containt zeroes, meaning we have to append some zeroes  to the original input
-	// so that both will produce the same hash
-	if len(wantChunks[len(wantChunks)-1]) != bufSz {
-		for i := len(wantChunks[len(wantChunks)-1]); i < bufSz; i++ {
-			wantChunks[len(wantChunks)-1] = append(wantChunks[len(wantChunks)-1][:], 0)
-		}
+func CollectFunc[I, O any](seq iter.Seq[I], f func(I) O) (o []O) {
+	for v := range seq {
+		o = append(o, f(v))
 	}
+	return
+}
 
-	// Hash the keys using SHA256 and add them to the slice to compare to hte keys from Keys()
-	for i := range wantChunks {
-		h := sha256.New()
-		h.Write(wantChunks[i])
-		s := binary.LittleEndian.Uint64(h.Sum(nil))
-		wantKeyHashes = append(wantKeyHashes, s)
+func TestHashKeysFromChunks(t *testing.T) {
+	tests := []struct {
+		name      string
+		hashFunc  func([]byte) uint64
+		word      string
```

Comment:
`word` isn't quite right naming here. Maybe calling it just `in` would be best?

PS: I assume you are aware of the renaming feature in VScode; so this should be quick to fix.